### PR TITLE
Remove ruler.drain-notification-queue-on-shutdown option, and make enabled it by default

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

Remove `-ruler.drain-notification-queue-on-shutdown` option, and enable it by default. We're doing this after successful rollout and usage of the feature in our production environment.

#### Checklist

- [na] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [na] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 * [CHANGE] Distributor: Replace `-distributor.retry-after-header.max-backoff-exponent` and `-distributor.retry-after-header.base-seconds` with `-distributor.retry-after-header.min-backoff` and `-distributor.retry-after-header.max-backoff` for easier configuration. #8694
 * [CHANGE] Ingester: increase the default inactivity timeout of active series (`-ingester.active-series-metrics-idle-timeout`) from `10m` to `20m`. #8975
 * [CHANGE] Distributor: Remove `-distributor.enable-otlp-metadata-storage` flag, which was deprecated in version 2.12. #9069
+* [CHANGE] Ruler: `-drain-notification-queue-on-shutdown` is now enabled by default, and no longer considered experimental.
```
Comment (pracucci) [inline CHANGELOG.md:30]:
```suggestion
* [CHANGE] Ruler: `-ruler.drain-notification-queue-on-shutdown` is now enabled by default, and no longer considered experimental.
```

**Review by pracucci — APPROVED** [review]
LGTM, but do we want to keep the flag at all? Should we just remove it and keep the feature always enabled?

**Review by pracucci — APPROVED** [review]

**Comment by pstibrany** [comment]
> LGTM, but do we want to keep the flag at all? Should we just remove it and keep the feature always enabled?

There's no rush in merging, we can check with @charleskorn who introduced the flag. (I'm only doing some cleanup of our internal release files, hence the PR)

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +27,7 @@
 * [CHANGE] Distributor: Replace `-distributor.retry-after-header.max-backoff-exponent` and `-distributor.retry-after-header.base-seconds` with `-distributor.retry-after-header.min-backoff` and `-distributor.retry-after-header.max-backoff` for easier configuration. #8694
 * [CHANGE] Ingester: increase the default inactivity timeout of active series (`-ingester.active-series-metrics-idle-timeout`) from `10m` to `20m`. #8975
 * [CHANGE] Distributor: Remove `-distributor.enable-otlp-metadata-storage` flag, which was deprecated in version 2.12. #9069
+* [CHANGE] Ruler: `-ruler.drain-notification-queue-on-shutdown` is now enabled by default, and no longer considered experimental.
```
Comment (tacole02) [inline CHANGELOG.md:30]:
```suggestion
* [CHANGE] Ruler: `-ruler.drain-notification-queue-on-shutdown` is enabled by default and no longer experimental.
```

**Review by tacole02 — APPROVED** [review]
Looks great. Thank you, @pstibrany!

**Comment by charleskorn** [comment]
> LGTM, but do we want to keep the flag at all? Should we just remove it and keep the feature always enabled?

I think we should just remove the flag and always enable draining the queue.

File: CHANGELOG.md (hunk 2 above)
Comment (pstibrany) [reply CHANGELOG.md:30]: Thanks. I've modified the changelog entry, since we have removed the option completely. PTAL.

**Comment by pstibrany** [comment]
I've removed the option completely.

**Review by pracucci — APPROVED** [review]
👍

**Pull request merged**